### PR TITLE
Updates regarding account level grid functionality for Date of Service Sorting

### PR DESCRIPTION
**Pull request by anudishCGI**
This PR contains the updates regarding the date of service sorting implementation on the account level grid

Impacted classes:
HCCostAccountController
HCCostAccountDAO

LWC Components:
ambulanceRecordsAccount
hospitalRecordsAccount
mspRecordsAccount
pharmacareRecordsAccount
continuingcareAccount
futurecareAccount

Could you please review @cgijeffolsen and @NataliaNikishina 
Thank You

### CODE REVIEW

File: TPL/force-app/main/default/classes/HCCCostControllerTest.cls [hunk 1]
```diff
@@ -398,10 +398,11 @@ public class HCCCostControllerTest {
         List<Healthcare_Cost__c> caseAssociation2 = new List<Healthcare_Cost__c>();
         String caseId = c2.Id;
         String exceptionMessage = null;   
+        String sortOrder = Label.TPL_Descending;
         try{
             insert new Healthcare_Cost__c[] {hcc1, hcc2, hcc3}; 
-            accountDetailWrapper1 = HCCostAccountController.getHealthcareCostsAmbulanceForAccount(acc.Id, filterAllRecords, pageNumber, pageSize);
-            accountDetailWrapper2 = HCCostAccountController.getHealthcareCostsAmbulanceForAccount(acc.Id, filterUncheckedRecords, pageNumber, pageSize);
+            accountDetailWrapper1 = HCCostAccountController.getHealthcareCostsAmbulanceForAccount(acc.Id, filterAllRecords, pageNumber, pageSize, sortOrder);
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCCostControllerTest.cls:404]: Sort Order has been added for each Account data grid as a parameter to return correctly sorted data.

File: TPL/force-app/main/default/classes/HCCostAccountDAO.cls [hunk 2]
```diff
@@ -1,63 +1,163 @@
...
-    public static List<Healthcare_Cost__c> getHCCostAccountRecords(String accId, String selectedFilterValue, Id recordTypeId, Integer startItemNumber, Integer pageSize, String fields){
-           
-        Integer itemsToRemove = startItemNumber - 2;
-        list<Healthcare_Cost__c> hccList = new list<Healthcare_Cost__c>();
-        String dateId = '';
-        String query = '';
-        Boolean costValue = false;
-        try{
-           
+public static List<Healthcare_Cost__c> getHCCostAccountRecords(String accId, String selectedFilterValue, Id recordTypeId, Integer startItemNumber, Integer pageSize, String fields, String sortOrder){
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCostAccountDAO.cls:2]: The data access methods have been updated to implement ascending or descending sort on date of service.

**Comment by anudishCGI** [comment]
HCCCostControllerTest class updated later to comply new changes in date sort code functionality.

File: TPL/force-app/main/default/classes/HCCostAccountDAO.cls [hunk 3]
```diff
@@ -1,63 +1,163 @@
...
-        }
-        catch(LimitException l){
-            System.debug(l.getMessage());
-        }
-        catch(ListException listexception){
-            System.debug(listexception.getMessage());
         }
-        catch(DmlException dml){
-            System.debug(dml.getMessage());
+        else{
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCostAccountDAO.cls:80]: The alternate branch was added due to the number of changes needed between ascending and descending. A review was completed to determine that the code would be larger if these branches were consolidated.

**Review by cgijeffolsen — APPROVED** [review]
@anudishCGI @NataliaNikishina Approved

**Review by NataliaNikishina — APPROVED** [review]
@anudishCGI approved
cc: @cgijeffolsen

**Pull request merged**